core-backend/common/src/funcs.rs

```diff
@@ -646,18 +649,19 @@ where
     pub fn out_of_gas(ctx: &mut R) -> Result<(), R::Error> {
         syscall_trace!("out_of_gas");
 
-        ctx.set_termination_reason(
-            ActorTerminationReason::Trap(TrapExplanation::GasLimitExceeded).into(),
-        );
+        let ext = ctx.ext_mut();
+        let current_counter = ext.current_counter();
+        log::trace!(target: "syscalls", "[out_of_gas] Current counter in global represents {current_counter:?}");
 
-        Err(R::unreachable_error())
-    }
-
-    pub fn out_of_allowance(ctx: &mut R) -> Result<(), R::Error> {
-        syscall_trace!("out_of_allowance");
+        if current_counter == CounterType::GasAllowance {
+            // We manually decrease it to 0 because global won't be affected
+            // since it didn't pass comparison to argument of `gas_charge()`
+            ext.decrease_to(0);
+        }
```

Comment:
There's nothing we need to solve.
That's normal situation because:
**previously**
GAS: 100
GAS ALLOWANCE: 90
- charge 95 -
GAS: 5
GAS ALLOWANCE: 90
- out_of_allowance() called -
- handled things on the backend cause in the end we operate only with gas counter: JournalNote::StopProcessing -